tests/utils/test_file.py

```diff
@@ -1,84 +1,53 @@
 import pytest
 import os
 import shutil
-from buildtest.utils.file import (
-    is_dir,
-    is_file,
-    create_file,
-    create_dir,
-    walk_tree,
-)
+import uuid
+from buildtest.utils.file import is_dir, is_file, create_dir, walk_tree, resolve_path
 from buildtest.exceptions import BuildTestError
 
 here = os.path.dirname(os.path.abspath(__file__))
 
 
-@pytest.mark.xfail(
-    reason="Test expected to fail when checking an obscure directory path",
-    raises=BuildTestError,
-)
 def test_checking_directory():
-    is_dir("/xxXXxxyyYYyyYYyyzzZZzzZZzZZZz")
+    dirname = str(uuid.uuid4())
+    assert not is_dir(dirname)
 
 
 @pytest.mark.xfail(
     reason="Test expected to fail when checking an obscure file", raises=BuildTestError
 )
 def test_checking_file():
-    is_file("/xXXXxxXXYyyyYYYyyYzZZZzZZZ")
-
-
-def test_directory_expansion():
-    is_dir("$HOME")
-    is_dir("~")
-
-    assert True is os.path.isdir(os.path.expanduser("~"))
-    assert True is os.path.isdir(os.path.expandvars("$HOME"))
+    file_name = str(uuid.uuid4())
+    assert not is_file(file_name)
 
+    file1 = "~/.profile"
+    file2 = "$HOME/.profile"
 
-def test_check_file():
-    is_file("~/.profile")
-    is_file("$HOME/.profile")
+    assert is_file(file1)
+    assert is_file(file2)
 
-    assert True is os.path.isfile(os.path.expanduser("~/.profile"))
-    assert True is os.path.isfile(os.path.expandvars("$HOME/.profile"))
-
-
-def test_create_file():
-    create_file("/tmp/a.txt")
-    create_file("$HOME/a.txt")
-    create_file("~/b.txt")
-
-    assert True is os.path.isfile("/tmp/a.txt")
-    # checking variable expansion
-    assert True is os.path.isfile(os.path.expandvars("$HOME/a.txt"))
-    # checking ~ expansion
-    assert True is os.path.isfile(os.path.expanduser("~/b.txt"))
-
-    os.remove("/tmp/a.txt")
-    os.remove(os.path.expandvars("$HOME/a.txt"))
-    os.remove(os.path.expanduser("~/b.txt"))
 
+def test_directory_expansion():
+    dir1 = "$HOME"
+    dir2 = "~"
 
-@pytest.mark.xfail(
-    reason="Expected Failure in creating file because lack of permission",
-    raises=OSError,
-)
-def test_fail_create_file():
-    create_file("/etc/a.txt")
+    assert is_dir(dir1)
+    assert is_dir(dir2)
 
 
 def test_create_dir():
-    create_dir("$HOME/a/b/c")
-    create_dir("~/x/y/z")
-    assert True is os.path.isdir(os.path.expandvars("$HOME/a/b/c"))
-    assert True is os.path.isdir(os.path.expanduser("~/x/y/z"))
-    shutil.rmtree(os.path.expandvars("$HOME/a/"))
-    shutil.rmtree(os.path.expanduser("~/x/"))
+    dir1 = os.path.join("$HOME", "a", "b", "c")
+    dir2 = os.path.join("~", "x", "y", "z")
+    create_dir(dir1)
+    create_dir(dir2)
+    assert is_dir(dir1)
+    assert is_dir(dir2)
+    shutil.rmtree(os.path.expandvars(os.path.join("$HOME", "a")))
```

Comment:
This shutil rmtree to HOME makes me VERY uncomfortable. A test should not be trying to modify anything outside of tmp_path.

Reply:
what would you like me to do in order to test shell and user expansion. Should i remove this test and just refactor this as part of ``resolve_path`` which just a simple assert without checking if ``create_dir`` can resolve path. The thing is ``create_dir`` doesn't use ``resolve_path`` because of the reason i mentioned in commit message https://github.com/shahzebsiddiqui/buildtest-framework/commit/2a8d6749580f2abad0c3e4c65a85b5a3d9e580e3

Reply:
Just to clarify these lines are needed instead of calling ``resolve_path`` since non-existent file will return ``None`` from ``resolve_path``. Since input to ``create_dir`` expects a non-existent file we can't use ``resolve_path``. We can expect ``create_dir`` to be called in this case with using ``buildtest get`` or even ``buildtest build --testdir`` where directory is auto-created. 

https://github.com/shahzebsiddiqui/buildtest-framework/blob/f8e34452871d53df2ca22b8cfb75eca1977af92b/buildtest/utils/file.py#L109-L111

i did shutil.rmtree to clean up after test creation. It may not be the best way to do it and am open to a better solution so long as we test shell and user expansion.

Without this test someone using ``buildtest build --testdir $HOME`` will result in a bug.

Reply:
I think it would make sense to create an environment variable relative to the tmp_dir created by pytest, and then expand that. We should absolutely not be venturing into an actual user home and touching anything.

Reply:
Let me see if i can use ``tmp_path`` and create a sub-directory i can only do shell expansion. User expansion is out of question since ``~`` refers to home directory. I dont have a way to test that other than what i have already.

Reply:
okay for now i am not testing directory expansion within ``tmp_path`` I am not really sure how to go about that. 
To keep it simple i just test directory creation with a sub-directory in ``tmp_path``. See https://github.com/shahzebsiddiqui/buildtest-framework/commit/e955909d73ab4391bc2ad5864f32425e5bbf30f9